include/envoy/common/platform.h

```diff
@@ -235,3 +235,17 @@ struct mmsghdr {
   unsigned int msg_len;
 };
 #endif
+
+
+#define SUPPORTS_PTHREAD_GETNAME_NP
+#ifdef __linux__
+#undef SUPPORTS_PTHREAD_GETNAME_NP
+#endif
+
+// https://android.googlesource.com/platform/bionic/+/master/docs/status.md
+// pthread_getname_np is introduced in API 26
+#ifdef __ANDROID_API__
+#if __ANDROID_API__ < 26
+#undef SUPPORTS_PTHREAD_GETNAME_NP
+#endif // __ANDROID_API__ < 26
```

Comment:
Are there versions of Android where `__ANDROID_API__` isn't defined? i.e. should we also just check for `__ANDROID__`?

Reply:
I'm assuming that `__ANDROID_API__` not being defined means we're not building for android. Is there a need to check for `__ANDROID__`?

Reply:
I was just suggesting we `#undef` support if `__ANDROID__` is `#define`d and then def it again if `__ANDROID_API__` _is_ present _and_ above 26.

Reply:
Well wondering if that's what we should do, since I wasn't sure about the ubiquity of `__ANDROID_API__`

Reply:
I think we can rely on this always being passed by bazel